src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs

```diff
@@ -248,29 +271,25 @@ int IDebugPendingBreakpoint2.Bind()
                         {
                             case enum_BP_LOCATION_TYPE.BPLT_CODE_FUNC_OFFSET:
                                 {
-                                    IDebugFunctionPosition2 functionPosition = HostMarshal.GetDebugFunctionPositionForIntPtr(_bpRequestInfo.bpLocation.unionmember2);
-                                    EngineUtils.CheckOk(functionPosition.GetFunctionName(out functionName));
+                                    EngineUtils.CheckOk(_functionPosition.GetFunctionName(out functionName));
                                     break;
                                 }
                             case enum_BP_LOCATION_TYPE.BPLT_CODE_CONTEXT:
                                 {
-                                    IDebugCodeContext2 codePosition = HostMarshal.GetDebugCodeContextForIntPtr(_bpRequestInfo.bpLocation.unionmember1);
-                                    if (!(codePosition is AD7MemoryAddress))
+                                    if (!(_codePosition is AD7MemoryAddress))
                                     {
                                         goto default;   // context is not from this engine
                                     }
-                                    codeAddress = ((AD7MemoryAddress)codePosition).Address;
+                                    codeAddress = ((AD7MemoryAddress)_codePosition).Address;
                                     break;
                                 }
                             case enum_BP_LOCATION_TYPE.BPLT_CODE_FILE_LINE:
                                 {
-                                    IDebugDocumentPosition2 docPosition = HostMarshal.GetDocumentPositionForIntPtr(_bpRequestInfo.bpLocation.unionmember2);
-
                                     // Get the name of the document that the breakpoint was put in
-                                    EngineUtils.CheckOk(docPosition.GetFileName(out documentName));
+                                    EngineUtils.CheckOk(_docPosition.GetFileName(out documentName));
```

Comment:
@gregg-miskelly I guess my question is in these calls where its GetFileName or GetRange, I should be caching that in the bind call also?

Reply:
I think we should use the cached info. Really this whole thing is a bit wrong in that we should really be asking GDB where the break point bound. But if we aren't going to do that, it is better to use the info we cached then to re-ask VS for it.